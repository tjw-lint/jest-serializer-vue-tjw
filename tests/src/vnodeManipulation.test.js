const fs = require('fs');

const helpers = require('../helpers.js');
const loadOptions = require('../../src/loadOptions.js');
const vnodeManipulation = require('../../src/vnodeManipulation.js');

import { shallowMount } from '@vue/test-utils';

describe('VnodeManipulation', () => {
  test('Deep clone fails', () => {
    helpers.mockSettings({
      addInputValues: true,
      verbose: false
    });

    const options = loadOptions(fs);
    const wrapper = shallowMount({
      name: 'fail',
      template: '<div></div>'
    });

    /**
     * A mock of lodash.clonedeep
     */
    function deepFake () {
      throw new Error('Max callstack exceed, oh no');
    }

    expect(vnodeManipulation(wrapper, options, deepFake))
      .toMatchSnapshot();
  });
});
