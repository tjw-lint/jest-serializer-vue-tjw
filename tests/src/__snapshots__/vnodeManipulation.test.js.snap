// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VnodeManipulation Deep clone fails 1`] = `<div></div>`;
